### PR TITLE
Remove boost from the dependencies as it is no longer used anyway

### PR DESCRIPTION
**Pull request by JanuszL**
Signed-off-by: Janusz Lisiecki <jlisiecki@nvidia.com>

#### Why we need this PR?
*Pick one, remove the rest*
- Remove boost from the dependencies as it is no longer used anyway

#### What happened in this PR?
*Fill relevant points, put NA otherwise. Replace anything inside []*
 - What solution was applied:
     Removes boost from the dependencies as it is no longer used anyway
 - Affected modules and functionalities:
     deps
 - Key points relevant for the review:
     NA
 - Validation and testing:
     CI
 - Documentation (including examples):
     NA

**JIRA TASK**: *[NA]*

### CODE REVIEW

File: dali/pipeline/data/types.h [hunk 1]
```diff
@@ -33,17 +33,7 @@
...
-// Workaround missing "is_trivially_copyable" in libstdc++ for g++ < 5.0.
-// We have to first include some standard library headers, so to have __GLIBCXX__ symbol,
-// and we have to exclude the specific version used in manylinux for our CI, because
-// clang always defines __GNUC__ to 4. __GLIBCXX__ is not a linear ordering based on
-// version of library but the date of the release so we can have 4.9.4 > 5.1.
-#if __GLIBCXX__ == 20150212 || (__cplusplus && __GNUC__ < 5 && !__clang__)
-#include <boost/type_traits/has_trivial_copy.hpp>
-#define IS_TRIVIALLY_COPYABLE(T) ::boost::has_trivial_copy<T>::value
-#else
 #define IS_TRIVIALLY_COPYABLE(T) std::is_trivially_copyable<T>::value
```
Comment (klecki) [inline dali/pipeline/data/types.h:36]: TBH, you could remove this macro as well.

Comment (JanuszL) [reply dali/pipeline/data/types.h:36]: done

**Comment by JanuszL** [comment]
!build

**Comment by JanuszL** [comment]
!build

**Pull request merged**